src/AppHost/Program.cs

```diff
@@ -43,8 +43,19 @@
 
 var blobStorage = storage.AddBlobs("eshopsupport-blobs");
 
-var pythonInference = builder.AddPythonUvicornApp("python-inference",
-    Path.Combine("..", "PythonInference"), port: 62394);
+var stage = builder.ExecutionContext.IsRunMode ? "base" : /* default stage */ null;
+var pythonInference = builder.AddDockerfile("python-inference", "../PythonInference", /* default dockerfile */ null, stage)
+    .WithHttpEndpoint(targetPort: 62394, env: "UVICORN_PORT")
+    .WithContainerRuntimeArgs("--gpus=all")
+    .WithLifetime(ContainerLifetime.Persistent);
```

Comment:
Does this definitely work? It does cause the Docker container to keep running even after AppHost is shut down. But each time I start the AppHost project using Ctrl+F5, it kills the previous Docker container and starts a new one, which seems to defeat the purpose of the feature.

Reply:
> Does this definitely work? It does cause the Docker container to keep running even after AppHost is shut down (albeit removing the port mapping, which is a bit inconvenient).

If you want stable ports inside and outside of Aspire for this resource you can pass the desired port numbers and disable the proxying on the endpoint.

> But then each time I start the AppHost project using Ctrl+F5, it kills the previous Docker container and starts a new one, which seems to defeat the purpose of the feature.

Definitely shouldn't be doing that and I wasn't seeing that locally. It will only do that if it detects changes to the container definition but I'm not sure of the exact variables it includes. @danegsta?

Reply:
By default we consider the container image name (or image sha in the case of building via dockerfile), ports, volume mounts, ports, environment, and the command/arguments used to launch the container.

The port mapping shouldn't be lost for a persistent container, but it generally won't show up in the Docker Desktop UI due to a seeming bug in how they report randomly assigned host ports (which we use). You have to inspect the container and look in the networking config section to see the real port, otherwise it just shows up as 0 in the other inspected port data and is missing from the UI.